### PR TITLE
[feat] Manifest based authorization - Authorization manifest and image hash separation

### PR DESCRIPTION
**Pull request by mhatrevi**

### CODE REVIEW

File: auth-manifest/gen/src/lib.rs [hunk 1]
```diff
@@ -39,6 +41,24 @@ pub struct AuthManifestGeneratorConfig {
...
 
     pub owner_man_key_info: Option<AuthManifestGeneratorKeyConfig>,
+}
+
+// /// Image Metadata Collection Image Generator Key Configuration
+// #[derive(Default, Clone)]
+// pub struct ImcGeneratorKeyConfig {
+//     pub priv_keys: Option<AuthManifestPrivKeys>,
+// }
+
```
Comment (swenson) [inline auth-manifest/gen/src/lib.rs:51]: Accidentally left in?

```suggestion
```

File: runtime/src/authorize_and_stash.rs [hunk 2]
```diff
@@ -62,20 +63,52 @@ impl AuthorizeAndStashCmd {
...
+            let mut auth_result = IMAGE_NOT_AUTHORIZED;
+            let mut fw_id_found = false;
+            for metadata_entry in auth_manifest_image_metadata_col
+                .image_metadata
+                .image_metadata_list
+                .iter()
+            {
+                if cmd.fw_id == metadata_entry.fw_id {
+                    fw_id_found = true;
+                    if cfi_launder(metadata_entry.digest) == cmd.measurement {
```
Comment (swenson) [inline runtime/src/authorize_and_stash.rs:95]: Are we not doing constant-time comparisons for digests?

Comment (swenson) [inline runtime/src/authorize_and_stash.rs:95]: We should use constant-time comparisons for user-provided hashes when possible, e.g., `constant_time_eq::constant_time_eq`.

**Review by swenson — APPROVED** [review]
LGTM, though I am a little concerned about constant-time comparisons. I'll do a separate investigation on that though.

File: api/src/mailbox.rs [hunk 3]
```diff
@@ -49,6 +49,9 @@ impl CommandId {
     // The authorization manifest set command.
     pub const SET_AUTH_MANIFEST: Self = Self(0x4154_4D4E); // "ATMN"
 
+    // The authorization manifest set command.
```
Comment (swenson) [inline api/src/mailbox.rs:52]:
```suggestion
    // The image metadata set command.
```

File: runtime/src/set_image_metadata.rs [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+
+Licensed under the Apache-2.0 license.
+
+File Name:
+
+    set_image_metadata.rs
+
+Abstract:
+
+    File contains AuthManifest mailbox command.
```
Comment (swenson) [inline runtime/src/set_image_metadata.rs:11]:
```suggestion
    File contains Set Image Metadata mailbox command.
```

**Review by nquarton — COMMENTED** [review]
This looks pretty good to me. I think this mainly needs an update in the runtime readme with the new command and its usage. Is there (or will there be) an update to the original "Measurement Manifest" doc that describes this feature?

**Pull request closed without merging**